tests/components/here_travel_time/test_sensor.py

```diff
@@ -0,0 +1,841 @@
+"""The test for the here_travel_time sensor platform."""
+import unittest
+from unittest.mock import patch
+import urllib
+
+import requests_mock
+
+from homeassistant.components.here_travel_time.sensor import (
+    ATTR_ATTRIBUTION, ATTR_DESTINATION_NAME, ATTR_DISTANCE, ATTR_DURATION,
+    ATTR_DURATION_WITHOUT_TRAFFIC, ATTR_ORIGIN_NAME, ATTR_ROUTE, CONF_MODE,
+    CONF_TRAFFIC_MODE, CONF_UNIT_SYSTEM, ICON_CAR, ICON_PEDESTRIAN,
+    ICON_PUBLIC, ICON_TRUCK, ROUTE_MODE_FASTEST, ROUTE_MODE_SHORTEST,
+    SCAN_INTERVAL, TRAFFIC_MODE_DISABLED, TRAFFIC_MODE_ENABLED,
+    TRAVEL_MODE_CAR, TRAVEL_MODE_PEDESTRIAN, TRAVEL_MODE_PUBLIC,
+    TRAVEL_MODE_TRUCK, UNIT_OF_MEASUREMENT)
+from homeassistant.const import ATTR_ICON
+from homeassistant.setup import setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import (
+    fire_time_changed, get_test_home_assistant, load_fixture)
+
+DOMAIN = 'sensor'
+
+PLATFORM = 'here_travel_time'
+
+APP_ID = 'test'
+APP_CODE = 'test'
+
+
+def _build_mock_url(origin, destination, modes, app_id, app_code, departure):
+    """Construct a url for HERE."""
+    base_url = (
+        "https://route.cit.api.here.com/routing/7.2/calculateroute.json?")
+    parameters = {
+        'waypoint0': origin,
+        'waypoint1': destination,
+        'mode': modes,
+        'app_id': app_id,
+        'app_code': app_code,
+        'departure': departure
+    }
+    url = base_url + urllib.parse.urlencode(parameters)
+    return url
+
+
+class TestHereTravelTimeSensor(unittest.TestCase):
```

Comment:
Please use standalone pytest test functions for the tests, not test classes.

If possible use pytest fixture functions to extract the patching and set up code from the tests.

Reply:
Could you point me to an example where standalone pytest test functions are used in combination with requests.get mocking? I am lost.